### PR TITLE
Add final demo: CI/CD pipeline using Docker and Github actions

### PR DESCRIPTION
**Pull request by JohanKJIP**
# Demo: CI/CD pipeline using Docker and Github actions

## Final submission links

**Youtube**: https://www.youtube.com/watch?v=Gc8SlsqPp-c

Subtitles can be enabled by clicking the subtitles button on youtube.

**Github**: https://github.com/JohanKJIP/ci-cd-demo

## Members

Johan von Hacht (johvh@kth.se)
David Johansson (davidjo2@kth.se)

## Topic

Demo of a CI/CD pipeline using Github actions.

## Details

We have implemented a CI/CD pipeline for a small website using Docker and Github actions. The demo shows how the pipeline works in action. 

We also provide the viewers with a take-home message. The repository is also available on Github so that the viewer can fork it if they want to experiment with the pipeline.

### CODE REVIEW

**Comment by jvikstrom** [comment]
I'll give feedback here

**Comment by Volminger** [comment]
Hey,
I will give feedback to you :)

**Comment by jadlers** [comment]
I'd like to give feedback to this demo :relaxed:

**Comment by javierron** [comment]
I will give feedback for this demo :)

**Comment by MindaugasVarkalys** [comment]
I would like to give feedback for this

**Comment by ashgreyship** [comment]
I'll give feedback for this demo.

**Comment by Goradux** [comment]
i would like to give feedback!

**Comment by sashikanthr** [comment]
I would like to give feedback

**Comment by sujon2100** [comment]
I will give feedback!

**Comment by sujon2100** [comment]
Here is the feedback:
The screencast looks great. In general, it has a good explanation by the subtitles and with the pointers on the screen. Just as a general suggestion to improve the screencast is instead of showing some video in the background, you could add some related video that you are talking about on the screen. Besides, Easter eggs could be added as a hidden field so that the viewers can easily click and reveal that wherever they want. You could also add take-home advice by giving your opinion that you think is related to the problem-solving. Otherwise, it's a great job. Wish you good luck!

**Comment by ashgreyship** [comment]
The introduction of the demo is interesting and useful. It is easy for the user to understand the advantages of using GitHub actions to create a CI/CD pipeline. However, The demo is too general. It is quite hard to follow the steps. The user may have difficulty replicating the demonstration. I recommend the author to discuss more details of every step in the screencast.

**Comment by jvikstrom** [comment]
This is really impressive visually. 
Is 0:13 stock footage or did you film it yourself? 

2:32 The change to `important_variable` is genius, I just found that incredibly funny.

The quality of this is incredible: it feels like some "official" marketing video that Github would do when releasing actions.

However I have two points you should look into: Firstly is no description on the youtube video as far as I can tell (but my Youtube page seems to be broken on this video). Secondly, even though the intro is fantastic, I think for this setting you should probably reduce the time you spend in the introduction. But that second point is really mostly down to my personal preference, I prefer it when they are quick to get to the point.

As you have some extra time to work with maybe consider digging in on the Github Actions yaml? Or at least the most important parts to set up your own Github Actions CI platform.

The take-home message is generally good: I do think you should try to rephrase it a bit (I think I get a bit annoyed at the "even more" in the end. But without that you probably need to rephrase it more to make it flow nice)

Great job on the demo, I'm honestly a bit stunned at the quality of it.

**Comment by Goradux** [comment]
Feedback:

Good length, clear title. The introduction makes it clear why the topic is relevant/useful to DevOps. Great addition that both of you participated in the video, unlike most of the other demos (us included). Also a great example you have decided to use :D (you get that Easter egg point). Although the complexity of the demo does not seem to be extremely hard, Github Actions is a very useful skill to have, and you guys did a great job showing it. Very visually pleasing, but maybe you have talked a tiny bit too slowly, although I had to speed you guys up to 1.2x, it can be benefitial for someone who prefers slower explanations. Great job, I think it is the best demo I have seen today (of the three I covered)!

**Comment by javierron** [comment]
- Really good demo, very clear and high quality voice and screen recording and editing.
- Content is very helpful and motivates to get started using GitHub Actions
----
- I would try to make the demo more accessible by giving more context, e.g.: Specify that the web server is deployed as a docker container, or where the YAML files should be created, etc.
- Provide the repository/YAML files for reference, and for the viewers to be able to experiment themselves.

**Comment by jadlers** [comment]
- Very nice B-roll and professional looking video introduction. Impressive!
- The audio is good and you are both talking clearly.
- A suggestion for something that could make it easier for the audience to
  follow is to quickly show an overview or say what you will be looking at
  or describing next.
    - For example [here](https://youtu.be/Gc8SlsqPp-c?t=51) (51s) you could
      say something along the lines of "Now we'll show how to define these
      actions".
    - Then again
      [here](https://www.youtube.com/watch?v=Gc8SlsqPp-c&feature=youtu.be&t=130)
      (2m10s): "Let's now see the GitHub actions in action"
    - You do it well [here](https://youtu.be/Gc8SlsqPp-c?t=183) (3m3s).
- As for the video quality it does not look fully HD at all times. Note that
  this comment comes from a person who is very picky when it comes to video
  quality and I don't mean that it makes the demo difficult to follow in any
  way. Just pointing out a minor possible enhancement to the demo. One
  obvious example is when looking closer at the file defining the test
  action [here](https://youtu.be/Gc8SlsqPp-c?t=59) (59s). The text is quite
  pixelated. For the most parts it might be resolvable by re-rendering the
  video with higher quality.
- A very minor comment is to, like in papers, speak out the abbreviation
  CI/CD the first time it's said. This might not necessary as it is a well
  known term for anyone in the field. Saying continuous
  integration/deployment once however wouldn't hurt.
- On a final note it looks like you've fulfilled all mandatory requirements
  for the demo, good job!

**Comment by MindaugasVarkalys** [comment]
The screencast is perfect! You did a great job with video editing. I really enjoyed your visuals as well as the background music. Your speech was clear and very easy to understand. However, I would prefer if you talk a bit faster. The remaining time you can use to explain your `yml` configuration file in more detail, so even the ones not familiar with Github Actions could learn something from your video. It's great that you explained the benefits of Github Actions and why it matters for Devops as well as included a Github repository. Overall, it's high quality demo.

**Comment by Volminger** [comment]
Really great demo, enjoyed watching it! It was entertaining and you showed great use of humor in your demo. The music and your voices work really well together, so it is clear to hear what you are saying. 

**Visual**: Although the visual side of this tutorial is already very impressive. I would suggest making the logs a bit larger at around 2:25 and increasing the zoom in the code around 2:28

**Take-home message**: Great use of the duck to deliver a great take-home message

**Technical**: You could improve upon this part by digging more into workflow file. Maybe explain a bit about GitHub secrets and how you use them in your workflow. You also mention that you publish your container on Docker Hub, maybe you could show us this update over at Docker Hub.

**GitHub**: It is great that you have a link to your GitHub repo and that this contain a great README on how one can run this yourself.

**Comment by sashikanthr** [comment]
Beginnning of the video is very impressive. It looks like a professional video for a consulting firm.  The narration is concise and clear with a nice ambient sound. The demo was easy to follow and clearly achieved the purpose of showing CI / CD using github actions.  like the simple way you demonstarted Corono clicker with just a change in a line of code. I also liked a clear take home message. The only minor drawback I see is for people who are not familiar with containers or yml configurations it might be a little difficult to follow. But again, the original purpose is well explained. 
Great work and Good luck! :)
